js/src/rive_advanced.mjs.d.ts

```diff
@@ -168,68 +372,140 @@ interface RiveOptions {
+     * This also applies new values to properties of objects on the Artboard according to the keys of the animation.
+     * This must be called after the `advance()` method of `LinearAnimationInstance`
+     *  
      * @param mix 0-1 the strength of the animation in the animations mix.
      */
     apply(mix: number): any;
-    // Deletes the backing Wasm animation instance
+    /**
+     * Deletes the underlying instance created via the WASM. It's important to clean up this instance when no longer in use
+     */
     delete(): void;
   }
-  
+
+  /**
+   * Rive class representing a StateMachine instance. Use this class to advance and control a particular state machine in the render loop (i.e scrub, grab state machine inputs, set up listener events, etc.).
+   *
+   * Important: Make sure to delete this instance when it's no longer in use via the `delete()` method. This deletes the underlying c++ reference and
+   * frees up the backing WASM object. This can be done in cases where the user navigates away from the page with this animation, the canvas is unmounted, etc.
+   */
   export declare class StateMachineInstance {
+    /**
+     * Create a new StateMachineInstance reference
+     * @param stateMachine - A StateMachineInstance retrieved via the Artboard (i.e `artboard.stateMachineByName('foo')`)
+     * @param artboard - The Artboard instance for this state machine
+     */
     constructor(stateMachine: StateMachineInstance, artboard: Artboard);
     get name(): string;
+    /**
+     * Returns the number of inputs associated with this state machine
+     * @returns Number of inputs
+     */
     inputCount(): number;
-    input(i: number): SMIInput; 
+    /**
+     * Returns the SMIInput at the given index
+     * @param i - Index to retrieve the state machine input at
+     * @returns SMIInput reference
+     */
+    input(i: number): SMIInput;
+    /**
+     * Advances/scrubs the StateMachineInstance by the set amount of seconds. Note that this does not apply changes to the properties of objects in the Artboard yet.
+     * @param sec - Scrub the state machine instance by a number of seconds
+     */
     advance(sec: number): any;
+    /**
+     * Returns the number of states changed while the state machine played
+     * @returns Number of states changed in the duration of the state machine played
+     */
     stateChangedCount(): number;
+    /**
+     * Returns the name of the state/animation transitioned to, given the index in the array of state changes in total
+     * @param i
+     * @returns Name of the state/animation transitioned to
+     */
     stateChangedNameByIndex(i: number): string;
 
-    // Call with coordinates in Artboard space
+    /**
+     * Registers a coordinate where the pointer down listener watches for in the Artboard space. Internally, Rive may advance a state machine if the listener coordinate is of interest at a given moment.
```

Comment:
Do we need to talk about state machine listeners here? (Are these things the user can register on?)

Or could we just say something as simple as "Notifies the state machine that the pointer has been pressed"?

Reply:
Yeah I like that verbiage - this one was weird to try and document